### PR TITLE
Email backend for SendGrid

### PR DESCRIPTION
**Pull request by shirishgoyal**
a) Add a new backend for SendGrid to remove hardcoded dependencies
b) Allow local settings to override email backend to use django's available backends if needed
c) Add console backend as default for local settings

### CODE REVIEW

**Comment by dmorina** [comment]
should we use `sendmail` for dev systems?

**Comment by dmorina** [comment]
lgtm

**Comment by shirishgoyal** [comment]
I think any email provider like console or file based is enough for dev system. 

Apart from that, backend should not limit developer's choice and we leave it to him/her to decide for local system.

File: crowdsourcing/emails.py [hunk 1]
```diff
@@ -84,4 +51,68 @@ def send_password_reset_email(email, host, reset_key):
                    ""<br><p>If you didn't request a password reset please click the following link: <br>" + \
                    '<a href="' + reset_url + '/0' + '">' + reset_url + '/0' + \
                    '</a><br><br> Greetings,<br> <strong>Daemo Team</strong>'
-    send_grid(to, subject, text_content, html_content)
+
+    send_mail(email_from, email_to, subject, text_content, html_content)
+
+# def send_activation_email_gmail(email, host, activation_key):
```
Comment (dmorina) [inline crowdsourcing/emails.py]: you can drop all this below

**Pull request merged**